### PR TITLE
Fixes file upload fails for files with zero size

### PR DESCRIPTION
**Pull request by dawoodkhan82**
# Description
Added some error handling for when file size is zero

Please include: 
* relevant motivation
* a summary of the change 
* which issue is fixed. 
* any additional dependencies that are required for this change.

Closes: #2778

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have added a short summary of my change to the CHANGELOG.md
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# A note about the CHANGELOG

Hello 👋 and thank you for contributing to Gradio!

All pull requests must update the change log located in CHANGELOG.md, unless the pull request is labeled with the "no-changelog-update" label.

Please add a brief summary of the change to the Upcoming Release > Full Changelog section of the CHANGELOG.md file and include
a link to the PR (formatted in markdown) and a link to your github profile (if you like). For example, "* Added a cool new feature by `[@myusername](link-to-your-github-profile)` in `[PR 11111](https://github.com/gradio-app/gradio/pull/11111)`".

If you would like to elaborate on your change further, feel free to include a longer explanation in the other sections.
If you would like an image/gif/video showcasing your feature, it may be best to edit the CHANGELOG file using the 
GitHub web UI since that lets you upload files directly via drag-and-drop.

### CODE REVIEW

**Comment by freddyaboulton** [comment]
Is this ready for review @dawoodkhan82 ?

File: CHANGELOG.md [hunk 1]
```diff
@@ -27,7 +27,8 @@ Progress indicator bar by [@aliabid94](https://github.com/aliabid94) in [PR 2750
 * Fixed bug where csv preview for DataFrame examples would show filename instead of file contents by [@freddyaboulton](https://github.com/freddyaboulton) in [PR 2877](https://github.com/gradio-app/gradio/pull/2877)
 * Fixed bug where an error raised after yielding iterative output would not be displayed in the browser by 
 [@JaySmithWpg](https://github.com/JaySmithWpg) in [PR 2889](https://github.com/gradio-app/gradio/pull/2889)
-* Fixed bug in `blocks_style` demo that was preventing it from launching by [@freddyaboulton](https://github.com/freddyaboulton) in [PR 2890](https://github.com/gradio-app/gradio/pull/2890)  
+* Fixed bug in `blocks_style` demo that was preventing it from launching by [@freddyaboulton](https://github.com/freddyaboulton) in [PR 2876](https://github.com/gradio-app/gradio/pull/2890)
```
Comment (freddyaboulton) [inline CHANGELOG.md:30]: This line is an accidental change?

File: gradio/processing_utils.py [hunk 2]
```diff
@@ -76,7 +76,10 @@ def get_extension(encoding: str) -> str | None:
         return "flac"
     elif type is None:
         return None
-    extension = mimetypes.guess_extension(type)
+    try:
+        extension = mimetypes.guess_extension(type)
+    except:
```
Comment (freddyaboulton) [inline gradio/processing_utils.py:81]: Can we avoid using a bare `except` clause here?

**Review by freddyaboulton — APPROVED** [review]
Looks good to me @dawoodkhan82 ! I would add a unit test and possibly avoid using a bare except clause

**Comment by abidlabs** [comment]
LGTM @dawoodkhan82, works great for both `UploadButton` and `File`.

**Review by abidlabs — APPROVED** [review]

**Comment by abidlabs** [comment]
Added a test & it's ready to merge

**Pull request merged**